src/ampycloud/data.py

```diff
@@ -93,16 +94,33 @@ def _cleanup_pdf(self, data: pd.DataFrame) -> pd.DataFrame:
         # Begin with a thorough inspection of the dataset
         data = utils.check_data_consistency(data, req_cols=self.DATA_COLS)
 
+        # By default we set NSC to false, NOTE: attrs is still experimental in pandas!
```

Comment:
well yes. As @fpavogt mentions, we could just set a `CeiloChunk` attribute. But then I think CeiloChunk is already to big which is why I'd like to keep the flag close to the data. And there it's the only way. I tried defining a specific `AmpycloudDataFrame` class, but it would lead to having to convert everything you read in with the pandas library. It's a bit hard to explain, we can chat if you need clarification.

Reply:
Maybe the idea of setting up an `AmpycloudDataFrame` could be added to the todo list ? Doing so could be a good way to stabilize the API in terms of the data input by further shielding the user from the inner-workings. And allow for exploring the potential of polars to replace pandas, which could/should lead to even better performances (cf #96) ? Such a replacement would however be a rather big surgery. I would advise against doing it right away/in one go.

Reply:
Yes! I'd like to look into polars at some point. But I do agree that this would be a lot of work.